Assets/Features/Telemetry Lap File/IdealTelemetryLapCreatorCorrector.cs

```diff
@@ -0,0 +1,61 @@
+﻿using UnityEngine;
+
+namespace Perrinn424.TelemetryLapSystem
+{
+    public class IdealTelemetryLapCreatorCorrector
+    {
+        private float dt;
+        private float time;
+        private float distanceOffset;
+
+        private CSVLine temp;
+        private CSVLine current;
+        private CSVLine previous;
+        private CSVLine corrected;
+
+        public int LineSector => temp.Sector;
+        public float[] CorrectedValues => corrected.Values;
+
+        public void ReadLine(string line)
+        {
+            temp.UpdateValues(line);
+        }
+        public IdealTelemetryLapCreatorCorrector(string [] headers, float dt)
+        {
+            temp = new CSVLine(headers);
+            current = new CSVLine(headers);
+            previous = new CSVLine(headers);
+            corrected = new CSVLine(headers);
+
+            this.dt = dt;
+            time = 0f;
+            distanceOffset = 0f;
+        }
+
+        public void Correct()
+        {
+            previous.UpdateValues(current);
+            current.UpdateValues(temp);
+            corrected.UpdateValues(current);
+
+            corrected.Time = time;
+            time += dt;
+
+            //if (!previous.HasValues) 
```

Comment:
if commented out, shall we remove these?

Reply:
I missed that. I'll remove it

Reply:
Removed at:
74728114a99a6c73ad6598f638ff716171181ca8